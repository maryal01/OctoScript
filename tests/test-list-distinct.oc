list<int> l1 = [1, 2, 3, 2, 4, 3, 5, 1, 1, 3];
list<int> l2 = l1.distinct();
print(string_of_list(l2));

list<float> l3 = [1.1, 2.2, 3.3, 2.2, 4.4, 3.3, 5.5, 1.1, 1.1, 3.3];
list<float> l4 = l3.distinct();
print(string_of_list(l4));

list<bool> l5 = [true, true, false, true, false];
list<bool> l6 = l5.distinct();
print(string_of_list(l6));

/*
list<string> l7 = ["a", "b", "a", "c", "c", "b", "a", "d", "b"];
list<string> l8 = l7.distinct();
print(string_of_list(l8));
*/