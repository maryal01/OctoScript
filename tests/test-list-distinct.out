[1, 2, 3, 4, 5][1.100000, 2.200000, 3.300000, 4.400000, 5.500000][true, false]